### PR TITLE
obs-filters: Fix building of noise reduction

### PR DESCRIPTION
**Pull request by pkviet**
### Description
If speex and rnnoise are disabled but nvafx is enabled, the noise reduction filter still needs to be built. This fixes the issue.

### Motivation and Context
Enable building of nvafx even if speex & rnnoise are disabled.
This is useful for minimal test builds.
Bug discovered by @RytoEX ; fix found by @RytoEX 

### How Has This Been Tested?
obs builds even when  speex & rnnoise are disabled, while nvafx is enabled.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-filters/CMakeLists.txt [hunk 1]
```diff
@@ -39,8 +35,8 @@ target_link_libraries(obs-filters PRIVATE OBS::libobs $<$<PLATFORM_ID:Windows>:O
 
 include(cmake/speexdsp.cmake)
 include(cmake/rnnoise.cmake)
-
```
Comment (RytoEX) [inline plugins/obs-filters/CMakeLists.txt:42]: Revert this line.

File: plugins/obs-filters/cmake/nvidia.cmake [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if(ENABLE_NVAFX)
+  target_sources(obs-filters PRIVATE noise-suppress-filter.c)
+  target_enable_feature(obs-filters LIBNVAFX_ENABLED HAS_NOISEREDUCTION)
+else()
+  target_disable_feature(obs-filters "NVIDIA Audio FX support")
+endif()
```
Comment (RytoEX) [inline plugins/obs-filters/cmake/nvidia.cmake:6]: End files with a newline.

File: plugins/obs-filters/cmake/nvidia.cmake [hunk 3]
```diff
@@ -0,0 +1,6 @@
+if(ENABLE_NVAFX)
+  target_sources(obs-filters PRIVATE noise-suppress-filter.c)
+  target_enable_feature(obs-filters LIBNVAFX_ENABLED HAS_NOISEREDUCTION)
```
Comment (RytoEX) [inline plugins/obs-filters/cmake/nvidia.cmake:3]: IIRC, `target_enable_feature` _requires_ a message, and the ARGN are optional things to pass to `target_compile_definitions`.
https://github.com/obsproject/obs-studio/blob/c521b23619f9938bc00de38d9ca15a3fe0061685/cmake/common/helpers_common.cmake#L67-L74

You either have to do:
```cmake
target_enable_feature(obs-filters "NVIDIA Audio FX migrations" LIBNVAFX_ENABLED HAS_NOISEREDUCTION)
```
or, if you don't want to call out the "feature" in the CMake configure printout:
```cmake
target_compile_definitions(obs-filters PRIVATE LIBNVAFX_ENABLED HAS_NOISEREDUCTION)
```

File: plugins/obs-filters/cmake/nvidia.cmake [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+else()
+  target_disable_feature(obs-filters "NVIDIA Audio FX support")
```
Comment (RytoEX) [inline plugins/obs-filters/cmake/nvidia.cmake:5]: If you choose to directly call `target_compile_definitions`, you don't need this `else()` branch.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: plugins/obs-filters/CMakeLists.txt (hunk 1 above)
Comment (pkviet) [reply plugins/obs-filters/CMakeLists.txt:42]: done

File: plugins/obs-filters/cmake/nvidia.cmake (hunk 3 above)
Comment (pkviet) [reply plugins/obs-filters/cmake/nvidia.cmake:3]: ok, have put it to private then

File: plugins/obs-filters/cmake/nvidia.cmake (hunk 2 above)
Comment (pkviet) [reply plugins/obs-filters/cmake/nvidia.cmake:6]: done

**Review by RytoEX — CHANGES_REQUESTED** [review]
Also, split the nv-filters change into its own commit so that it can be appropriately prefixed.

File: plugins/obs-filters/cmake/nvidia.cmake (hunk 4 above)
Comment (pkviet) [reply plugins/obs-filters/cmake/nvidia.cmake:5]: right

**Comment by pkviet** [comment]
> Also, split the nv-filters change into its own commit so that it can be appropriately prefixed.

will do

File: plugins/obs-filters/cmake/nvidia.cmake (hunk 3 above)
Comment (RytoEX) [reply plugins/obs-filters/cmake/nvidia.cmake]: Just noting that the "PRIVATE" is not the distinction here. The main point is `target_enable_feature` vs. `target_compile_definitions`.

**Comment by RytoEX** [comment]
Are you going to re-push the nv-filters commit?

**Comment by pkviet** [comment]
> Are you going to re-push the nv-filters commit?

nope, it was more of a clean up so the commit would not fit in the PR scope

**Comment by pkviet** [comment]
> > Are you going to re-push the nv-filters commit?
> 
> nope, it was more of a clean up so the commit would not fit in the PR scope

following internal discussion repushed the nv-filters commit

**Review by RytoEX — CHANGES_REQUESTED** [review]
Nit, second commit message:
* Remov -> Remove

**Review by RytoEX — APPROVED** [review]
This seems fine. It fixes the issue that I raised. The only quirk is that if you only build with `NVAFX_ENABLED`, and the other options disabled, then you can add a Noise Suppression filter without actually being able to view/change any properties. In practice, no one should be building for distribution this way though. Maybe in some future major release we can remove the migrations.

**Comment by pkviet** [comment]
> This seems fine. It fixes the issue that I raised. The only quirk is that if you only build with `NVAFX_ENABLED`, and the other options disabled, then you can add a Noise Suppression filter without actually being able to view/change any properties. In practice, no one should be building for distribution this way though. Maybe in some future major release we can remove the migrations.

Yeah. It seems unavoidable since the porting can't happen if the noise reduction filter is disabled.
It'll have to wait next version to remove the migration code and fix this.

**Pull request merged**